src/main/java/com/example/flightplanner/Services/FlightService.java

```diff
@@ -0,0 +1,111 @@
+package com.example.flightplanner.Services;
+
+import com.example.flightplanner.HelperClasses.AddFlightRequest;
+import com.example.flightplanner.HelperClasses.Airport;
+import com.example.flightplanner.HelperClasses.Flight;
+import com.example.flightplanner.HelperClasses.SearchFlightsRequest;
+import com.example.flightplanner.Repositories.FlightRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+@Service
+public class FlightService {
```

Comment:
Ok lielākā problēma, kuru es te redzu - Tu vērsies pa taisno pie flightRepository.flights, bet tā darīt nedrīkst. Tas, kas ir jāizdara - tas ko rakstīju jau iepriekš - iekš repository tos flights liec privātu. Un darbošanās ap to listu notiek tikai iekš repository. Ko tas nozīmē - to, ka, piemēram delete metodes saturs pilnībā jāpārnes uz repository, tā ka te paliek tikai:

```
    public void deleteFlight(int id) {
        flightRepository.deleteFlight(id);
    }
```
Tāpat add metodei jābūt šādi:
`flightRepository.add(flight);` nevis šādi: `flightRepository.flights.add(flight);`

Pa lielam service var palikt tikai kā karkass, kurā ir šādas metodes:

```
    public void deleteFlight(int id) {
        flightRepository.deleteFlight(id);
    }

    public Flight fetchFlightById(int id) {
        return flightRepository.fetchFlightById(id);
    }

    public List<Flight> searchFlight(SearchFlightsRequest request) {
        return flightRepository.searchFlight(request);
    }
```

Reply:
Paldies par padomiem, visu salaboju!